Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py

```diff
@@ -1571,54 +1579,60 @@ def long_running_loop():
-        if port_mapping:
-            if ':' in port_mapping:
-                port = int(port_mapping.split(':')[1])
+        try:
+            port_mapping: str = PARAMS.get('longRunningPort', '')
+            port: int
+            if port_mapping:
+                if ':' in port_mapping:
+                    port = int(port_mapping.split(':')[1])
+                else:
+                    port = int(port_mapping)
             else:
-                port = int(port_mapping)
-        else:
-            raise ValueError('No port mapping was provided')
-        Thread(target=channel_mirror_loop, daemon=True).start()
-        demisto.info('Started channel mirror loop thread')
-
-        ssl_args = dict()
-
-        if certificate and private_key:
-            certificate_file = NamedTemporaryFile(delete=False)
-            certificate_path = certificate_file.name
-            certificate_file.write(bytes(certificate, 'utf-8'))
-            certificate_file.close()
-            ssl_args['certfile'] = certificate_path
-
-            private_key_file = NamedTemporaryFile(delete=False)
-            private_key_path = private_key_file.name
-            private_key_file.write(bytes(private_key, 'utf-8'))
-            private_key_file.close()
-            ssl_args['keyfile'] = private_key_path
-
-            demisto.info('Starting HTTPS Server')
-        else:
-            demisto.info('Starting HTTP Server')
+                raise ValueError('No port mapping was provided')
+            Thread(target=channel_mirror_loop, daemon=True).start()
+            demisto.info('Started channel mirror loop thread')
+
+            ssl_args = dict()
+
+            if certificate and private_key:
+                certificate_file = NamedTemporaryFile(delete=False)
+                certificate_path = certificate_file.name
+                certificate_file.write(bytes(certificate, 'utf-8'))
+                certificate_file.close()
+                ssl_args['certfile'] = certificate_path
+
+                private_key_file = NamedTemporaryFile(delete=False)
+                private_key_path = private_key_file.name
+                private_key_file.write(bytes(private_key, 'utf-8'))
+                private_key_file.close()
+                ssl_args['keyfile'] = private_key_path
+
+                demisto.info('Starting HTTPS Server')
+            else:
+                demisto.info('Starting HTTP Server')
 
-        server = WSGIServer(('0.0.0.0', port), APP, **ssl_args)
-        server.serve_forever()
-    except Exception as e:
-        if certificate_path:
-            os.unlink(certificate_path)
-        if private_key_path:
-            os.unlink(private_key_path)
-        demisto.error(f'An error occurred in long running loop: {str(e)}')
-        raise ValueError(str(e))
+            server = WSGIServer(('0.0.0.0', port), APP, **ssl_args)
+            server.serve_forever()
+        except Exception as e:
+            error_message = str(e)
+        finally:
```

Comment:
I think in the finally you want to stop the server.  So you probably want to add something of the sort:
```
if server:
    server.stop()
```